File: app/src/app/domain/socialProjectVolunteerType.ts
export enum SocialProjectVolunteerType {
    MANAGER = "MANAGER",
	VOLUNTEER = "VOLUNTEER",
	CANDIDATE = "CANDIDATE",
	INVITED = "INVITED",
	NO_VOLUNTEER = "NO_VOLUNTEER"
}

export namespace SocialProjectVolunteerType {

	export function getSocialProjectVolunteerTypeName(socialProjectVolunteerType: SocialProjectVolunteerType): string {
		switch (socialProjectVolunteerType) {
			case SocialProjectVolunteerType.MANAGER:
				return "Gerente do projeto";
			case SocialProjectVolunteerType.VOLUNTEER:
				return "Voluntário";
			case SocialProjectVolunteerType.CANDIDATE:
				return "Candidato";
			case SocialProjectVolunteerType.INVITED:
				return "Convidado";
			case SocialProjectVolunteerType.NO_VOLUNTEER:
				return "Não voluntário";		
			default:
				return "";
		}
	}

	export function getSocialProjectVolunteerTypeColor(socialProjectVolunteerType: SocialProjectVolunteerType): string {
		switch (socialProjectVolunteerType) {
			case SocialProjectVolunteerType.MANAGER:
				return "purple";
			case SocialProjectVolunteerType.VOLUNTEER:
				return "cyan";
			case SocialProjectVolunteerType.CANDIDATE:
				return "green";
			case SocialProjectVolunteerType.INVITED:
				return "orange";
			case SocialProjectVolunteerType.NO_VOLUNTEER:
				return "red";		
			default:
				return "red";
		}
	}
}

File: api/src/main/java/br/com/furb/domain/SocialProjectVolunteerType.java
package br.com.furb.domain;

import lombok.Getter;

public enum SocialProjectVolunteerType {

	MANAGER("Gestor de projeto voluntário", "Você foi adicionado como gestor no projeto social %s, parabéns!"),
	VOLUNTEER("Voluntário em projeto social", "Você foi adicionado como voluntário no projeto social %s, parabéns!"),
	CANDIDATE("Candidatura em projeto social", "Você foi adicionado como candidato no projeto social %s. Os gestores estão analisando seu perfil."),
	INVITED("Convite para projeto social", "Você foi convidado para participar como voluntário no projeto social %s. Clique neste link para mais detalhes!"),
	NO_VOLUNTEER("Sem mensagem", "Sem mensagem");

	SocialProjectVolunteerType(String notificationTitle, String notificationMessage) {
		this.notificationTitle = notificationTitle;
		this.notificationMessage = notificationMessage;
	}

	public boolean isActive() {
		return this.equals(MANAGER) || this.equals(VOLUNTEER);
	}

	@Getter
	private String notificationTitle;
	@Getter
	private String notificationMessage;
}
